spanner/spansql/parser.go

```diff
@@ -2714,11 +2714,15 @@ func (p *parser) parseLit() (Expr, *parseError) {
 		p.back()
 		return p.parseArrayLit()
 	case tok.caseEqual("DATE"):
-		p.back()
-		return p.parseDateLit()
+		if !p.sniff("=") && !p.sniff(",") && !p.sniff("FROM") && !p.sniff("AS") {
```

Comment:
I think this should be implemented using a positive look-ahead instead of a negative-look ahead. In other words: If the `DATE` keyword is followed by a string literal, then it is a date literal. Otherwise it is not. The current implementation does not work for for example the following valid query:

```sql
SELECT UNIX_DATE(DATE "2008-12-25")
```

Also, adding additional exceptions to the current if-not list will not solve that permanently, as there might be changes to the query syntax that would add additional keywords (i.e. not only `AS` and `FROM`) that could follow an identifier.

Reply:
Agreeed.

Fixed at 007bcf2db5643fe597ae34cf5b0b9411b76fbd36